chatkit-core/src/main/kotlin/com/pusher/chatkit/CurrentUser.kt

```diff
@@ -233,45 +348,90 @@ class CurrentUser(
             callback: (Result<Int, Error>) -> Unit
     ) = makeCallback({ syncCurrentUser.sendMessage(roomId, messageText, attachment) }, callback)
 
+    /**
+     * Send a [messageText] to the [roomId]. The [callback] will be called with a [Result] when the
+     * operation is complete. If the operation was successful you will receive an [Int] which
+     * contains the message id, or an [Error] informing you of what went wrong.
+     */
     fun sendSimpleMessage(
             roomId: String,
             messageText: String,
             callback: (Result<Int, Error>) -> Unit
     ) = makeCallback({ syncCurrentUser.sendSimpleMessage(roomId, messageText) }, callback)
 
+    /**
+     * @see [sendSimpleMessage]
+     */
     fun sendSimpleMessage(
             room: Room,
             messageText: String,
             callback: (Result<Int, Error>) -> Unit
     ) = makeCallback({ syncCurrentUser.sendSimpleMessage(room, messageText) }, callback)
 
+    /**
+     * Sends some message [parts] to the [room]. For more information on what the [parts] can be check
+     * out the [docs][https://pusher.com/docs/chatkit/reference/android].
+     * The [callback] will be called with a [Result] when the operation is complete. If the
+     * operation was successful you will receive an [Int] which contains the message id, or an
+     * [Error] informing you of what went wrong.
```

Comment:
"Sends some message [parts] to the [room]" seems subtly, but confusingly different from "Send a message, which comprises several [parts], to a [room]"

I think this is a commonly used method where we should preferably provide a fairly complete example for reference as well as offering a link to the full docs.

Reply:
Okay - I already link to the docs that we have but I agree that we could also have an example. I've added a `/todo` to add that so I can get back to your other feedback and come back to this.

Reply:
To create a URL Message Part, what "type" should I provide -
`NewPart.Url("http://www.pusher.com", "text/plain"))` - is this what I'm supposed to do?

Reply:
the type of the media which is hosted at the link. For `pusher.com` it's going to be `text/html`, but the URL message part is intended for linking to media attachments, so it seems an unlikely example. `NewPart.Url("https://example.com/uploads/myphoto.png", "image/png")` might be more realistic, or `NewPart.Url("https://example.com/uploads/logs-to-share.txt", "text/plain")`